test/test_worker_types.py

```diff
@@ -1,318 +1,468 @@
-            {},
-            Exception,
-            id="wrong workflows format",
-        ),
-        pytest.param(
-            {"bitrise": {"app": "foo", "workflows": [{"nope": ["wut"]}]}},
-            {},
-            Exception,
-            id="wrong workflows format2",
-        ),
-        pytest.param(
-            {"bitrise": {"app": "foo", "workflows": [{"foo": {"bar": ["oops"]}}]}},
-            {},
-            Exception,
-            id="wrong workflows format3",
-        ),
-        pytest.param(
-            {"bitrise": {"app": "some-app", "workflows": ["bar", "baz"]}},
-            {},
-            {
-                "payload": {
-                    "global_params": {
-                        "branch": "default",
-                        "branch_repo_owner": "http://example.com/head/repo",
-                        "commit_hash": "abcdef",
-                    },
-                },
-                "scopes": [
-                    "foo:bitrise:app:some-app",
-                    "foo:bitrise:workflow:bar",
-                    "foo:bitrise:workflow:baz",
-                ],
-                "tags": {"worker-implementation": "scriptworker"},
+import mozilla_taskgraph.worker_types  # noqa - trigger payload_builder registration
+
+
+@pytest.fixture
+def build_payload(make_graph_config, make_transform_config, parameters):
+    graph_config = make_graph_config(
+        extra_config={"scriptworker": {"scope-prefix": "foo"}}
+    )
+
+    def inner(
+        name: str,
+        worker: dict,
+        extra_params: Optional[Dict] = None,
+        raises: Optional[Exception] = None,
+    ) -> Optional[Tuple[Dict, Dict]]:
+        extra_params = extra_params or {}
+
+        worker.setdefault("implementation", name)
+
+        task = {"worker": worker}
+        task_def = {"tags": {}}
+
+        parameters.update(extra_params)
+        config = make_transform_config(params=parameters, graph_cfg=graph_config)
+
+        payload_builder = payload_builders[name]
+
+        if inspect.isclass(raises) and issubclass(raises, Exception):
+            with pytest.raises(raises):
+                validate_schema(payload_builder.schema, worker, "schema error")
+                payload_builder.builder(config, task, task_def)
+        else:
+            validate_schema(payload_builder.schema, worker, "schema error")
+            payload_builder.builder(config, task, task_def)
+
+            print("Dumping task_def for copy/paste:")
+            pprint(task_def, indent=2)
+            return task, task_def
+
+    return inner
+
+
+def test_bitrise_missing(build_payload):
+    build_payload("scriptworker-bitrise", {}, raises=Exception)
+
+
+def test_bitrise_missing_app(build_payload):
+    build_payload(
+        "scriptworker-bitrise", {"bitrise": {"workflows": ["foo"]}}, raises=Exception
+    )
+
+
+def test_bitrise_missing_workflows(build_payload):
+    build_payload("scriptworker-bitrise", {"bitrise": {"app": "foo"}}, raises=Exception)
```

Comment:
This is aside from the issue at hand, but I have to say that these tests looks much nicer as written compared to the parametrized version of them.